src/GovUk.Education.ExploreEducationStatistics.Data.Services/TableBuilderResultSubjectMetaService.cs

```diff
@@ -23,77 +27,96 @@ public class TableBuilderResultSubjectMetaService : AbstractTableBuilderSubjectM
+                {
+                    var stopwatch = Stopwatch.StartNew();
+                    stopwatch.Start();
 
-            var stopwatch = Stopwatch.StartNew();
-            stopwatch.Start();
+                    var filters = GetFilters(observations);
 
-            var filters = GetFilters(observations);
+                    _logger.LogTrace("Got Filters in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
+                    stopwatch.Restart();
 
-            _logger.LogTrace("Got Filters in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
-            stopwatch.Restart();
+                    var footnotes = GetFootnotes(observations, query);
 
-            var footnotes = GetFootnotes(observations, query);
+                    _logger.LogTrace("Got Footnotes in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
+                    stopwatch.Restart();
 
-            _logger.LogTrace("Got Footnotes in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
-            stopwatch.Restart();
+                    var indicators = GetIndicators(query);
 
-            var indicators = GetIndicators(query);
+                    _logger.LogTrace("Got Indicators in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
+                    stopwatch.Restart();
 
-            _logger.LogTrace("Got Indicators in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
-            stopwatch.Restart();
+                    var locations = GetObservationalUnits(observations);
 
-            var locations = GetObservationalUnits(observations);
+                    _logger.LogTrace("Got Observational Units in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
+                    stopwatch.Restart();
 
-            _logger.LogTrace("Got Observational Units in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
-            stopwatch.Restart();
+                    var timePeriodRange = GetTimePeriodRange(observations);
 
-            var timePeriodRange = GetTimePeriodRange(observations);
+                    _logger.LogTrace("Got Time Periods in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
+                    stopwatch.Stop();
 
-            _logger.LogTrace("Got Time Periods in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
-            stopwatch.Stop();
+                    return new TableBuilderResultSubjectMetaViewModel
+                    {
+                        Filters = filters,
+                        Footnotes = footnotes,
+                        Indicators = indicators,
+                        Locations = locations,
+                        PublicationName = subject.Release.Publication.Title,
+                        SubjectName = subject.Name,
+                        TimePeriodRange = timePeriodRange
+                    };
+                });
+        }
 
-            return new TableBuilderResultSubjectMetaViewModel
+        private async Task<Either<ActionResult, Subject>> CheckSubjectExists(Guid subjectId)
```

Comment:
As per previous comment, could we replace this method with `persistenceHelper.CheckEntityExists<Subject>(subjectId)`?

Reply:
Done